processor/src/dtos/operations/payment-componets.dto.ts

```diff
@@ -1,25 +1,42 @@
 import { Static, Type } from '@sinclair/typebox';
 
+const DropinType = Type.Enum({
+  EMBEDDED: 'embedded',
+  HPP: 'hpp',
+  EXPRESS: 'express',
+});
+
+export const SupportedPaymentDropinsData = Type.Object({
+  type: DropinType,
+});
+
 export const SupportedPaymentComponentsData = Type.Object({
   type: Type.String(),
+  subtypes: Type.Optional(Type.Array(Type.String())),
 });
 
 /**
  * Supported payment components schema.
  *
  * Example:
  * {
- *  "components": [
- *    {
- *      "type": "card"
- *    },
- *    {
- *      "type": "applepay"
- *    }
- *  ]
+ *   "dropins": [
+ *     {
+ *       "type": "embedded"
+ *     }
+ *   ],
+ *   "components": [
+ *     {
+ *       "type": "card"
+ *     },
+ *     {
+ *       "type": "applepay"
+ *     }
+ *   ]
  * }
  */
 export const SupportedPaymentComponentsSchema = Type.Object({
+  dropins: Type.Array(SupportedPaymentDropinsData),
```

Comment:
Please, notice that both arrays are mandatories. The idea is to force to return both and let empty the array if dropin or components are not supported.
The alternative is to make both optional. 

Please, let me know if this should be changed.

Reply:
To chime in there, personally I like the current approach. An empty array already indicates that it does not support anything. If the array becomes optional and can thus be not be present or undefined the caller needs to handle those scenarios as well in the code. While they essentially (empty array or not present/defined) mean the same thing.